src/main/java/org/apache/mesos/curator/CuratorStateStore.java

```diff
@@ -137,18 +138,20 @@ public void clearFrameworkId() throws StateStoreException {
     }
 
     @Override
-    public Protos.FrameworkID fetchFrameworkId() throws StateStoreException {
+    public Optional<Protos.FrameworkID> fetchFrameworkId() throws StateStoreException {
         try {
             logger.debug("Fetching FrameworkID from '{}'", fwkIdPath);
             byte[] bytes = curator.fetch(fwkIdPath);
             if (bytes.length > 0) {
-                return Protos.FrameworkID.parseFrom(bytes);
+                return Optional.of(Protos.FrameworkID.parseFrom(bytes));
             } else {
                 throw new StateStoreException(String.format(
                         "Failed to retrieve FrameworkID in '%s'", fwkIdPath));
             }
+        } catch (KeeperException.NoNodeException e) {
+            logger.warn("No FrameworkId found.");
+            return Optional.empty();
```

Comment:
Ah, ok, that makes sense. I misread the `NoNodeException` as an unexpected error itself. Is `curator` a part of this repo? I like the idea of standardizing these semantics across the board, so it might be appropriate for `CuratorPersister::fetch` to return an `Optional` itself if its returning an empty value is a normal part of proceedings here.